pyuvdata/mwa_corr_fits.py

```diff
@@ -0,0 +1,235 @@
+            warnings.warn('only a subset of coarse channels are being processed')
+            
+        #read in the data!
+        #read into a data array with dimensions (time, frequency, baselines/pols)
+        #and a times array holds the unix timestamp from each hdu
+        #TODO: fix this to deal with a frequency or time subset
+        data_dump=np.zeros((self.Ntimes, self.Nfreqs, self.Nbls*self.Npols),dtype=np.complex)
+        times_dump=np.zeros(self.Ntimes)
+        #get a file from '00'
+        for file in file_dict['00']:
+            #get the course channel index from the file name
+            course=int(file.split('_')[-2][-2:])
+            with fits.open(file,memmap=True,do_not_scale_image_data=True) as hdu_list:
+                end_time=len(hdu_list)
+                #get number of fine channels
+                fine=hdu_list[1].header['NAXIS2']
+                freq_ind=(course-1)*fine
+                for i in range(1,end_time):
+                    #this takes data from real to complex numbers
+                    data_dump[i-1,freq_ind:freq_ind+fine,:]=hdu_list[i].data[:,0::2]+1j*hdu_list[i].data[:,1::2]
+                    times_dump[i-1]=(hdu_list[i].header['TIME']+hdu_list[i].header['MILLITIM']/1000)
+        start_time=end_time
+        for file in file_dict['01']:
+            #get the course channel index from the file name
+            course=int(file.split('_')[-2][-2:])
+            with fits.open(file,memmap=True,do_not_scale_image_data=True) as hdu_list:
+                end_time=len(hdu_list)
+                #get number of fine channels
+                fine=hdu_list[1].header['NAXIS2']
+                freq_ind=(course-1)*fine
+                for i in range(1,end_time):
+                    #this takes data from real to complex numbers
+                    data_dump[start_time+i-2,freq_ind:freq_ind+fine,:]=hdu_list[i].data[:,0::2]+1j*hdu_list[i].data[:,1::2]
+                    times_dump[start_time+i-2]=(hdu_list[i].header['TIME']+hdu_list[i].header['MILLITIM']/1000)
+        #TODO: read in flag files
+
+        #TODO: reindex the data_dump!
+        #TODO: convert times_dump to julian dates
+    
+            
+        
+        
+        
+        
+        
+        #TODO:
+        self.baseline_array=np.zeros(self.Nblts, dtype=np.complex)
+        #TODO:
+        self.ant_1_array=np.zeros(self.Nblts)
+        #TODO:
+        self.ant_2_array=np.zeros(self.Nblts)
+        #TODO:
+        self.data_array=np.zeros((self.Nblts, self.Nspws, self.Nfreqs, self.Npols))
+        #TODO:
+        self.flag_array=np.zeros((self.Nblts, self.Nspws, self.Nfreqs, self.Npols))
+        #TODO:
+        self.lst_array=np.zeros(self.Nblts)
+        #TODO:
+        self.nsample_array=np.zeros((self.Nblts, self.Nspws, self.Nfreqs, self.Npols))
+        #TODO
+        self.polarization_array=np.zeros(self.Npols)
+        #TODO ask Bryna what is the shape of this?
+        #self.telescope_location
```

Comment:
It's a 3 element vector giving the location of the telescope in ECEF coordinates. We have utility methods to convert from lat/lon/alt if you have those.